### PR TITLE
[OCPCLOUD-1815]: Modernize tooling

### PR DESCRIPTION
**Pull request by odvarkadaniel**
This commit introduces modernization of the approach to update auxiliary tooling to be more consistent such as in Control Plane Machine Set repository. We align envtest, controller-gen and other tooling as well as in the CPMS repo.

### CODE REVIEW

File: pkg/controllers/cache.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ func (n *objectWatcher) watch(ctx context.Context, obj client.Object) error {
 
 	// Add an event handler that only allows events through for the correct object name
 	// Since the informer is namespace bound, this should limit the events from this event handler to a single resource.
-	informer.AddEventHandler(&eventToChannelHandler{
+	_, _ = informer.AddEventHandler(&eventToChannelHandler{
```
Comment (lobziik) [inline pkg/controllers/cache.go:84]: i suppose `AddEventHandler` returns an error, can we propagate this instead suppressing?

File: go.mod [hunk 2]
```diff
@@ -1,53 +1,44 @@
...
 	github.com/asaskevich/govalidator v0.0.0-20210307081110-f21760c49a8d
 	github.com/go-logr/logr v1.2.3
 	github.com/golangci/golangci-lint v1.49.0
-	github.com/onsi/ginkgo/v2 v2.6.0
-	github.com/onsi/gomega v1.24.0
-	github.com/openshift/api v0.0.0-20230111143458-54592eea5539
-	github.com/openshift/library-go v0.0.0-20230112164258-24668b1349e6
-	github.com/spf13/cobra v1.5.0
+	github.com/onsi/ginkgo/v2 v2.7.0
+	github.com/onsi/gomega v1.24.2
```
Comment (lobziik) [inline go.mod:10]: gomega 1.25 was already released

**Review by lobziik — CHANGES_REQUESTED** [review]
Can we please exclude k8s dependencies bump from this pr?

Also, I wondering about XML reporting. The last time I experimented with it, it was broken (only one suite out of dozen was covered in this report). Could we check what will happen in case some test failing in a suite that is not covered by junit report.

**Comment by odvarkadaniel** [comment]
> Can we please exclude k8s dependencies bump from this pr?
> 
> Also, I wondering about XML reporting. The last time I experimented with it, it was broken (only one suite out of dozen was covered in this report). Could we check what will happen in case some test failing in a suite that is not covered by junit report.

I can separate the k8s deps into another PR.

AFAIK, this was broken in ginkgov2 v2.6.x, but already solved in v2.7.0.

**Comment by lobziik** [comment]
>AFAIK, this was broken in ginkgov2 v2.6.x, but already solved in v2.7.0.

IIRC this was coverage.

**Comment by odvarkadaniel** [comment]
> > AFAIK, this was broken in ginkgov2 v2.6.x, but already solved in v2.7.0.
> 
> IIRC this was coverage.

I will investigate it.

File: pkg/controllers/cache.go (hunk 1 above)
Comment (odvarkadaniel) [reply pkg/controllers/cache.go]: This change will be covered in an upcoming PR for k8s packages update. This is not an issue with the old version k8s having deps with older version golangci-lint.

File: go.mod (hunk 2 above)
Comment (odvarkadaniel) [reply go.mod]: Do you prefer me to update it in this PR or have it in the PR for k8s packages update?

File: go.mod [hunk 3]
```diff
@@ -1,53 +1,44 @@
...
-	github.com/stretchr/testify v1.8.0
+	github.com/stretchr/testify v1.8.1
 	gopkg.in/gcfg.v1 v1.2.3
 	gopkg.in/ini.v1 v1.66.6
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.25.2
-	k8s.io/apiextensions-apiserver v0.25.1
-	k8s.io/apimachinery v0.25.2
-	k8s.io/client-go v0.25.2
+	k8s.io/api v0.26.1
```
[the remark this answers is not in the document]
Comment (odvarkadaniel) [reply go.mod]: Rebase this PR. Separate PR will come for the k8s bump.

File: go.mod (hunk 2 above)
Comment (lobziik) [reply go.mod]: Here will be ok i think

**Comment by JoelSpeed** [comment]
/approve

I'm ok with this and i think increasing the test time is an ok addition, I suspect changing the tooling introduces some additional overhead within the test containers so CPU limitations are probably making these tests a little slower

**Review by lobziik — COMMENTED** [review]
>Also, I wondering about XML reporting. The last time I experimented with it, it was broken (only one suite out of dozen was covered in this report). Could we check what will happen in case some test failing in a suite that is not covered by junit report.

Here is the ginkgo issue: https://github.com/onsi/ginkgo/issues/1115
I'm ok to merge it as is, but we need to track JUnit reporting issue somewhere.

Will hold this for now, @JoelSpeed feel free to unhold, if we do not want to block this due to complications around reporting.

/lgtm
/hold

**Comment by JoelSpeed** [comment]
/hold cancel

I'd rather we get this merged and revisit later for the JUnit issue

**Pull request merged**